### ISSUE TITLE
NoMethodError - undefined method `start_with?' for true:TrueClass

### ISSUE THREAD

**Issue by maddy9859** [CocoaPods/CocoaPods#2923]
### Report

* What did you do?

* What did you expect to happen?

* What happened instead?

### Stack

```
   CocoaPods : 0.35.0
        Ruby : ruby 2.0.0p481 (2014-05-08 revision 45883) [universal.x86_64-darwin14]
    RubyGems : 2.0.14
        Host : Mac OS X 10.10.1 (14B25)
       Xcode : 6.1.1 (6A2008a)
         Git : git version 1.9.3 (Apple Git-50)
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib
Repositories :
               SafetyChangerLogin - git
               SafetyChangerVersionCheck -git
```

### Plugins

```
cocoapods-plugins : 0.3.2
cocoapods-trunk   : 0.4.1
cocoapods-try     : 0.4.2
```

### Error

```
NoMethodError - undefined method `start_with?' for true:TrueClass
/Library/Ruby/Gems/2.0.0/gems/cocoapods-core-0.35.0/lib/cocoapods-core/specification/linter/analyzer.rb:86:in `block (2 levels) in validate_file_patterns'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-core-0.35.0/lib/cocoapods-core/specification/linter/analyzer.rb:85:in `each'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-core-0.35.0/lib/cocoapods-core/specification/linter/analyzer.rb:85:in `block in validate_file_patterns'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-core-0.35.0/lib/cocoapods-core/specification/linter/analyzer.rb:77:in `each'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-core-0.35.0/lib/cocoapods-core/specification/linter/analyzer.rb:77:in `validate_file_patterns'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-core-0.35.0/lib/cocoapods-core/specification/linter/analyzer.rb:20:in `analyze'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-core-0.35.0/lib/cocoapods-core/specification/linter.rb:125:in `block (2 levels) in perform_all_specs_analysis'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-core-0.35.0/lib/cocoapods-core/specification/linter.rb:120:in `each'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-core-0.35.0/lib/cocoapods-core/specification/linter.rb:120:in `block in perform_all_specs_analysis'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-core-0.35.0/lib/cocoapods-core/specification/linter.rb:119:in `each'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-core-0.35.0/lib/cocoapods-core/specification/linter.rb:119:in `perform_all_specs_analysis'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-core-0.35.0/lib/cocoapods-core/specification/linter.rb:51:in `lint'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-core-0.35.0/lib/cocoapods-core/source/health_reporter.rb:88:in `lint_spec'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-core-0.35.0/lib/cocoapods-core/source/health_reporter.rb:53:in `block (2 levels) in analyze'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-core-0.35.0/lib/cocoapods-core/source/health_reporter.rb:50:in `each'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-core-0.35.0/lib/cocoapods-core/source/health_reporter.rb:50:in `block in analyze'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-core-0.35.0/lib/cocoapods-core/source/health_reporter.rb:49:in `each'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-core-0.35.0/lib/cocoapods-core/source/health_reporter.rb:49:in `analyze'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.35.0/lib/cocoapods/command/repo.rb:130:in `block in run'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.35.0/lib/cocoapods/command/repo.rb:122:in `each'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.35.0/lib/cocoapods/command/repo.rb:122:in `run'
/Library/Ruby/Gems/2.0.0/gems/claide-0.7.0/lib/claide/command.rb:271:in `run'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.35.0/lib/cocoapods/command.rb:45:in `run'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.35.0/bin/pod:43:in `<top (required)>'
/usr/bin/pod:23:in `load'
/usr/bin/pod:23:in `<main>'
```

**Comment by timonroe** [comment]
@maddy9859 hi.  I'm getting the same error.  You said you were 'adding the wrong source'.  Are you talking about the 'source' or 'source_files' in the podspec or something else?

**Comment by segiddins** [comment]
@timonroe can you please share your podspec?

**Comment by timonroe** [comment]
@segiddins  I'm trying to create a private pod, here's the podspec:

Pod::Spec.new do |s|
  s.name         = "MCAppLog"
  s.version      = "1.0.0"
  s.summary      = "Monroe Consulting Group common component - MCAppLog."
  s.description  = <<-DESC
                   MCAppLog: Monroe Consulting Group common component.
                   Common components used for building iOS apps.
                   DESC
  s.homepage     = "http://monroegolf.com"
  s.license      = 'MIT'
  s.author       = { "Tim Monroe" => "timonroecg@gmail.com" }
  s.source       = { :git => "/users/timonroe/.cocoapods/repos/mcapplog/.git", :tag => "1.0.0" }

  s.platform     = :ios, '8.1'
  s.ios.deployment_target = '8.1'
  s.requires_arc = true
  
  s.frameworks   = 'Foundation'

  s.source_files = 'MCAppLog/MCAppLog.h', 'MCAppLog/MCAppLog.m'
end

**Comment by timonroe** [comment]
@segiddins  Here's what I ran to add my private repo to the cocoapods installation:
pod repo add MCAppLog /users/timonroe/desktop/development/mccommoncomponents/mcapplog/.git
cd ~/.cocoapods/repos/MCAppLog
pod repo lint

Running lint gave me the error:
### Stack

```
   CocoaPods : 0.35.0
        Ruby : ruby 2.0.0p481 (2014-05-08 revision 45883) [universal.x86_64-darwin14]
    RubyGems : 2.4.5
        Host : Mac OS X 10.10.1 (14B25)
       Xcode : 6.1.1 (6A2008a)
         Git : git version 1.8.4
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ c1f98324f18d0482ac9f3d72b2adaa24e283b4c3
               MCAppLog - /users/timonroe/desktop/development/mccommoncomponents/mcapplog/.git @ 0f653d1c2f31cfc177d9f89f02da6229e1afa639
```

### Plugins

```
cocoapods-plugins : 0.3.1
cocoapods-trunk   : 0.4.1
cocoapods-try     : 0.4.3
```

### Error

```
NoMethodError - undefined method `start_with?' for true:TrueClass
/Library/Ruby/Gems/2.0.0/gems/cocoapods-core-0.35.0/lib/cocoapods-core/specification/linter/analyzer.rb:86:in `block (2 levels) in validate_file_patterns'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-core-0.35.0/lib/cocoapods-core/specification/linter/analyzer.rb:85:in `each'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-core-0.35.0/lib/cocoapods-core/specification/linter/analyzer.rb:85:in `block in validate_file_patterns'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-core-0.35.0/lib/cocoapods-core/specification/linter/analyzer.rb:77:in `each'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-core-0.35.0/lib/cocoapods-core/specification/linter/analyzer.rb:77:in `validate_file_patterns'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-core-0.35.0/lib/cocoapods-core/specification/linter/analyzer.rb:20:in `analyze'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-core-0.35.0/lib/cocoapods-core/specification/linter.rb:125:in `block (2 levels) in perform_all_specs_analysis'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-core-0.35.0/lib/cocoapods-core/specification/linter.rb:120:in `each'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-core-0.35.0/lib/cocoapods-core/specification/linter.rb:120:in `block in perform_all_specs_analysis'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-core-0.35.0/lib/cocoapods-core/specification/linter.rb:119:in `each'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-core-0.35.0/lib/cocoapods-core/specification/linter.rb:119:in `perform_all_specs_analysis'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-core-0.35.0/lib/cocoapods-core/specification/linter.rb:51:in `lint'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-core-0.35.0/lib/cocoapods-core/source/health_reporter.rb:88:in `lint_spec'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-core-0.35.0/lib/cocoapods-core/source/health_reporter.rb:53:in `block (2 levels) in analyze'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-core-0.35.0/lib/cocoapods-core/source/health_reporter.rb:50:in `each'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-core-0.35.0/lib/cocoapods-core/source/health_reporter.rb:50:in `block in analyze'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-core-0.35.0/lib/cocoapods-core/source/health_reporter.rb:49:in `each'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-core-0.35.0/lib/cocoapods-core/source/health_reporter.rb:49:in `analyze'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.35.0/lib/cocoapods/command/repo.rb:130:in `block in run'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.35.0/lib/cocoapods/command/repo.rb:122:in `each'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.35.0/lib/cocoapods/command/repo.rb:122:in `run'
/Library/Ruby/Gems/2.0.0/gems/claide-0.7.0/lib/claide/command.rb:271:in `run'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.35.0/lib/cocoapods/command.rb:45:in `run'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.35.0/bin/pod:43:in `<top (required)>'
/usr/bin/pod:23:in `load'
/usr/bin/pod:23:in `<main>'

**Comment by segiddins** [comment]
@timonroe should be fixed by https://github.com/CocoaPods/Core/pull/216

**Comment by velyan** [comment]
@timonroe Hi, you might get this error if your .podspec file is not located in the root of your repo.

**Comment by timonroe** [comment]
@segiddins  I just installed 36 Beta 1 and the problem still exists so assuming your changes aren't in there.  Do you know which version I can rollback to and get around this?

**Comment by segiddins** [comment]
@timonroe I'll need to investigate to be able to answer that, sorry.

**Comment by timonroe** [comment]
@velyan  Hi, my podspec file is unfortunately located in the root.

**Comment by timonroe** [comment]
@segiddins  I installed version 0.34.0 and no longer get this error.

**Comment by segiddins** [comment]
Thanks for the info @timonroe, I'll need to dig in to find the issue.

**Comment by timonroe** [comment]
@segiddins  Hi.  I'm now trying to use my private pod in a project and get the following error when running 'pod install'

Resolving dependencies of `Podfile`
Resolving dependencies for target `Pods' (iOS 8.1)
  - MCAppLog (= 1.0.0)
[!] An unexpected version directory `Base.lproj` was encountered for the `/Users/timonroe/.cocoapods/repos/MCAppLog/MCAppLog` Pod in the `MCAppLog` repository.

Here's what my podfile looks like:

source 'https://github.com/CocoaPods/Specs.git'
source '/users/timonroe/desktop/development/mccommoncomponents/mcapplog/.git'
platform :ios, '8.1'
pod 'MCAppLog', '1.0.0'

Any help with this is greatly appreciated.

**Comment by segiddins** [comment]
That's a different issue. 

-Samuel E. Giddins

On Jan 13, 2015, at 5:37 PM, Tim Monroe <notifications@github.com> wrote:

@segiddins Hi. I'm now trying to use my private pod in a project and get the following error when running 'pod install'

Resolving dependencies of Podfile
Resolving dependencies for target `Pods' (iOS 8.1)

MCAppLog (= 1.0.0) [!] An unexpected version directory Base.lproj was encountered for the /Users/timonroe/.cocoapods/repos/MCAppLog/MCAppLog Pod in the MCAppLog repository.
Here's what my podfile looks like:

source 'https://github.com/CocoaPods/Specs.git'
source '/users/timonroe/desktop/development/mccommoncomponents/mcapplog/.git'
platform :ios, '8.1'
pod 'MCAppLog', '1.0.0'

Any help with this is greatly appreciated.

**Comment by timonroe** [comment]
@segiddins   Following up just in case someone else runs into this...
The problem was caused by the structure of the project's folders.  The default structure of an XCode project will not work.  Pod install chokes on the Base.lproj folder if it's at the same level in the folder structure as the files used in the pod.  To fix it, I rearranged the folders/files so they looked like:
MCAppLog.podspec
\MCAppLog
    - MCAppLog.h
    - MCAppLog.m
\Demo
    \Base.lproj
    - AppDelegate.h
    - AppDelegate.m
    - ...
